### PR TITLE
Client and test fixes

### PR DESCRIPTION
**Pull request by edaniels**

### CODE REVIEW

File: src/client.js [hunk 1]
```diff
@@ -477,31 +477,30 @@ export class StitchClient {
...
-            return this._do(resource, method, options);
-          });
-        }
-
-        fetchArgs.headers.Authorization = `Bearer ${token}`;
-        return this._fetch(url, fetchArgs, resource, method, options);
-      }
-
-      fetchArgs.headers.Authorization = `Bearer ${token}`;
+    if (options.noAuth) {
```
Comment (edaniels) [inline src/client.js]: easier to read

File: test/client/http_service.test.js [hunk 2]
```diff
@@ -76,23 +76,23 @@ describe('Executing http service functions', () => {
...
+          await expect(service.get({ url: 'invalidurl' })).rejects.toMatchObject({
             name: 'StitchError',
             code: 'FunctionExecutionError',
             response: { status: 400 }
           });
         });
 
         it('should fail when an invalid url datatype is supplied', async() => {
-          expect(service.get({ url: 10281995 })).rejects.toMatchObject({
+          await expect(service.get({ url: 10281995 })).rejects.toMatchObject({
```
Comment (jsflax) [inline test/client/http_service.test.js]: [nit] I tend to place await directly in front of the promise declaration, as opposed to the outermost declaration.

File: src/client.js (hunk 1 above)
Comment (jsflax) [reply src/client.js]: 👍

File: test/client/http_service.test.js (hunk 2 above)
Comment (edaniels) [reply test/client/http_service.test.js]: This seemed to be the style the jest docs use

**Pull request merged**